File: src/main/java/com/extendedae_plus/common/registry/menu/host/linkLabel/HostCardChannel.java
package com.extendedae_plus.common.registry.menu.host.linkLabel;

import appeng.api.implementations.menuobjects.ItemMenuHost;
import appeng.menu.locator.ItemMenuHostLocator;
import com.extendedae_plus.common.registry.dataComponent.DataChannelCard;
import com.extendedae_plus.common.registry.item.upgradeCard.CardChannel;
import com.extendedae_plus.common.wireless.linkApi.Label;
import net.minecraft.world.entity.player.Player;

public class HostCardChannel extends ItemMenuHost<CardChannel> implements HostLabelLink {
    public HostCardChannel(CardChannel item, Player player, ItemMenuHostLocator locator) {
        super(item, player, locator);
    }

    @Override
    public Label.Data getLabelData() {
        return DataChannelCard.getLabel(this.getItemStack());
    }

    @Override
    public boolean setLabelData(Label.Data label, boolean force) {
        DataChannelCard.setLabel(this.getItemStack(), label, true);
        return true;
    }
}

File: src/main/kotlin/com/extendedae_plus/common/registry/menu/host/link/HostCardChannel.kt
package com.extendedae_plus.common.registry.menu.host.link

import appeng.api.implementations.menuobjects.ItemMenuHost
import appeng.menu.locator.ItemMenuHostLocator
import com.extendedae_plus.common.registry.dataComponent.DataChannelCard
import com.extendedae_plus.common.registry.item.upgradeCard.CardChannel
import com.extendedae_plus.common.wireless.linkApi.Label
import net.minecraft.world.entity.player.Player

class HostCardChannel(
    item: CardChannel, player: Player, locator: ItemMenuHostLocator
) : ItemMenuHost<CardChannel>(item, player, locator), HostLabelLink {
    override val labelData: Label.Data
        get() = DataChannelCard.getLabel(this.itemStack)

    override fun setLabelData(label: Label.Data, force: Boolean): Boolean {
        DataChannelCard.setLabel(this.itemStack, label, true)
        return true
    }
}
